fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/EntityDatatableChecksWritePlatformServiceImpl.java

```diff
@@ -105,10 +105,11 @@ public CommandProcessingResult createCheck(final JsonCommand command) {
                     throw new EntityDatatableCheckAlreadyExistsException(entity, status, datatableName);
                 }
             } else {
-                if (entity.equals("m_loan")) {
+                EntityTables entityTable = EntityTables.fromName(entity);
+                if (EntityTables.LOAN.equals(entityTable)) {
```

Comment:
minor: enum equality can be checked by identity

Reply:
Done